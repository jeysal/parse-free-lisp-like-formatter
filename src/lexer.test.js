const tokenize = require('./lexer');

test('recognizes all regular token types', () => {
  expect(
    tokenize(`
      (lambda id1 &id2
        (= true 123))
  `),
  ).toMatchSnapshot();
});

test('recognizes line comments', () => {
  expect(
    tokenize(`
      (a ;comment
      b)
  `),
  ).toMatchSnapshot();
});

test('recognizes block comments', () => {
  expect(
    tokenize(`
      (a #|com
      ment|# b)
  `),
  ).toMatchSnapshot();
});

test('generates tokens for empty lines', () => {
  expect(
    tokenize(`
      (a

      b)
    `),
  ).toMatchSnapshot();
});

test('treats a line as empty even if it contains whitespace', () => {
  expect(
    tokenize(`
      (a
\t
      b)
    `),
  ).toMatchSnapshot();
});

test('throws if the code includes an illegal char', () => {
  expect(() => tokenize('(a # b)')).toThrowError();
});
