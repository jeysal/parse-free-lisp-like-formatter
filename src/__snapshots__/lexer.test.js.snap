// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates tokens for empty lines 1`] = `
Array [
  Object {
    "type": "leftPar",
    "value": "(",
  },
  Object {
    "type": "identifier",
    "value": "a",
  },
  Object {
    "type": "emptyLine",
    "value": "",
  },
  Object {
    "type": "identifier",
    "value": "b",
  },
  Object {
    "type": "rightPar",
    "value": ")",
  },
]
`;

exports[`recognizes all regular token types 1`] = `
Array [
  Object {
    "type": "leftPar",
    "value": "(",
  },
  Object {
    "type": "keyword",
    "value": "lambda",
  },
  Object {
    "type": "identifier",
    "value": "id1",
  },
  Object {
    "type": "prefix",
    "value": "&",
  },
  Object {
    "type": "identifier",
    "value": "id2",
  },
  Object {
    "type": "leftPar",
    "value": "(",
  },
  Object {
    "type": "operator",
    "value": "=",
  },
  Object {
    "type": "boolLiteral",
    "value": "true",
  },
  Object {
    "type": "numLiteral",
    "value": "123",
  },
  Object {
    "type": "rightPar",
    "value": ")",
  },
  Object {
    "type": "rightPar",
    "value": ")",
  },
]
`;

exports[`recognizes block comments 1`] = `
Array [
  Object {
    "type": "leftPar",
    "value": "(",
  },
  Object {
    "type": "identifier",
    "value": "a",
  },
  Object {
    "type": "blockComment",
    "value": "#|com
      ment|#",
  },
  Object {
    "type": "identifier",
    "value": "b",
  },
  Object {
    "type": "rightPar",
    "value": ")",
  },
]
`;

exports[`recognizes line comments 1`] = `
Array [
  Object {
    "type": "leftPar",
    "value": "(",
  },
  Object {
    "type": "identifier",
    "value": "a",
  },
  Object {
    "type": "lineComment",
    "value": ";comment",
  },
  Object {
    "type": "identifier",
    "value": "b",
  },
  Object {
    "type": "rightPar",
    "value": ")",
  },
]
`;

exports[`treats a line as empty even if it contains whitespace 1`] = `
Array [
  Object {
    "type": "leftPar",
    "value": "(",
  },
  Object {
    "type": "identifier",
    "value": "a",
  },
  Object {
    "type": "emptyLine",
    "value": "",
  },
  Object {
    "type": "identifier",
    "value": "b",
  },
  Object {
    "type": "rightPar",
    "value": ")",
  },
]
`;
